x-pack/legacy/plugins/siem/server/lib/detection_engine/routes/signals/open_close_signals_route.ts

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import Hapi from 'hapi';
+import { DETECTION_ENGINE_SIGNALS_STATUS_URL } from '../../../../../common/constants';
+import { SignalsRequest } from '../../alerts/types';
+import { setSignalsStatusSchema } from '../schemas';
+import { ServerFacade } from '../../../../types';
+import { transformError } from '../utils';
+
+export const setSignalsStatusRouteDef: Hapi.ServerRoute = {
+  method: 'POST',
+  path: DETECTION_ENGINE_SIGNALS_STATUS_URL,
+  options: {
+    tags: ['access:siem'],
+    validate: {
+      options: {
+        abortEarly: false,
+      },
+      payload: setSignalsStatusSchema,
+    },
+  },
+  async handler(request: SignalsRequest, headers) {
+    const { signal_ids: signalIds, query, status } = request.payload;
+    const { callWithRequest } = request.server.plugins.elasticsearch.getCluster('data');
+    if (!callWithRequest) {
+      return headers.response().code(404);
+    }
+    if (status) {
```

Comment:
If status is sent in as `null | undefined` it looks like you don't return anything?

I think that Hapi will return a `500` internal server error and a nasty message if you don't explicitly return a value from a handler. If the `status` isn't sent in, I would return a `400` Boom to the user with a message that status is required.

Reply:
Ah yeah that would be a problem. I'll update the route and tests.

Reply:
Just pushed a commit to remove `null | undefined` as it's not present on the Signals interface https://github.com/elastic/kibana/blob/master/x-pack/legacy/plugins/siem/server/lib/types.ts#L78